urla5230a98e3_Dhana_Krishnasamy_InMemoryJMS_tgz-pJ8-com_dhana_broker_demo_AppJ8/src/com/dhana/broker/MockMessage.java

```diff
@@ -0,0 +1,69 @@
+package com.dhana.broker;
+
+import com.dhana.broker.Message;
+
+
+public class MockMessage implements Message {
+
+	String message;
+
+	// 1 second
+	long expiration = System.currentTimeMillis() + (3 * 1000);
+
+	public MockMessage(String message) {
+		this.message = message;
+		//this();
+		byte c='w';
+		char b=(char)c++;
+	}
+
+	public String getMessage() {
+		return this.message;
+	}
+
+	public void setMessage(String message) {
+		this.message = message;
+
+	}
+
+	public Message clone() {
+		return new MockMessage(this.message);
+	}
+
+	public boolean equals(Object that) {
+		if (!(that instanceof MockMessage))
+			return false;
+		String thatMessage = ((MockMessage) that).getMessage();
+		if (this.message == null)
+			return thatMessage == null;
+
+		return (this.message == thatMessage)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.